packages/twenty-front/src/modules/object-record/record-filter/utils/turnObjectDropdownFilterIntoQueryFilter.ts

```diff
@@ -24,6 +24,128 @@ export type ObjectDropdownFilter = Omit<Filter, 'definition'> & {
   };
 };
 
+const applyEmptyFilters = (
+  operand: ViewFilterOperand,
+  correspondingField: Pick<Field, 'id' | 'name'>,
+  objectRecordFilters: RecordGqlOperationFilter[],
+  filterType:
+    | 'URL'
+    | 'Address'
+    | 'String'
+    | 'Number'
+    | 'Date'
+    | 'Relation'
+    | 'CompositeString' = 'String',
+) => {
+  let emptyRecordFilter: RecordGqlOperationFilter = {};
+
+  switch (filterType) {
+    case 'String':
+      emptyRecordFilter = {
+        or: [
+          { [correspondingField.name]: { ilike: '' } as StringFilter },
+          { [correspondingField.name]: { is: 'NULL' } as StringFilter },
+        ],
+      };
+      break;
+    case 'CompositeString':
+      // eslint-disable-next-line no-case-declarations
+      const fullNameFilters = generateILikeFiltersForCompositeFields(
+        '',
+        correspondingField.name,
+        ['firstName', 'lastName'],
+      );
+      emptyRecordFilter = {
+        not: {
+          or: fullNameFilters,
+        },
+      };
+      break;
+    case 'URL':
+      emptyRecordFilter = {
+        or: [
+          { [correspondingField.name]: { url: { ilike: '' } } as URLFilter },
+          {
+            [correspondingField.name]: { url: { is: 'NULL' } } as URLFilter,
+          },
+        ],
+      };
+      break;
+    case 'Address':
+      emptyRecordFilter = {
+        or: [
```

Comment:
Shouldn't it be an and ? Because it feels like an address field is empty when all fields are empty no ?